h3.go

```diff
@@ -68,6 +68,44 @@ const (
 	RadsToDegs = 180.0 / math.Pi
 )
 
+// Error codes.
+var (
+	ErrFailed                = errors.New("the operation failed")
+	ErrDomain                = errors.New("argument was outside of acceptable range")
+	ErrLatLngDomain          = errors.New("latitude or longitude arguments were outside of acceptable range")
+	ErrResolutionDomain      = errors.New("resolution argument was outside of acceptable range")
+	ErrCellInvalid           = errors.New("H3Index cell argument was not valid")
+	ErrDirectedEdgeInvalid   = errors.New("H3Index directed edge argument was not valid")
+	ErrUndirectedEdgeInvalid = errors.New("H3Index undirected edge argument was not valid")
+	ErrVertexInvalid         = errors.New("H3Index vertex argument was not valid")
+	ErrPentagon              = errors.New("pentagon distortion was encountered")
+	ErrDuplicateInput        = errors.New("duplicate input was encountered in the arguments")
+	ErrNotNeighbors          = errors.New("H3Index cell arguments were not neighbors")
+	ErrRsolutionMismatch     = errors.New("H3Index cell arguments had incompatible resolutions")
+	ErrMemoryAlloc           = errors.New("necessary memory allocation failed")
+	ErrMemoryBounds          = errors.New("bounds of provided memory were not large enough")
+	ErrOptionInvalid         = errors.New("mode or flags argument was not valid")
+
+	errMap = map[C.uint]error{
```

Comment:
Think we should opt for `C.uint32_t` just to match what's explicitly in the C code... I think on the Go side they're the same, but seems like it'd be nice to match explicitly. https://github.com/uber/h3/blob/e8b16611b582275bdb9815bc5a80e0559a811758/src/h3lib/include/h3api.h.in#L79

Reply:
Done